runtime/dart_vm_initializer.cc

```diff
@@ -0,0 +1,66 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/runtime/dart_vm_initializer.h"
+
+#include <mutex>
+
+#include "flutter/fml/logging.h"
+#include "flutter/fml/trace_event.h"
+
+// Tracks whether Dart has been initialized and if it is safe to call Dart
+// APIs. This mutex must be held during calls to Dart_Initialize and
+// Dart_Cleanup so that Dart API calls on other threads can be protected
+// against a concurrent call to Dart_Cleanup.
+static std::mutex gDartInitializerMutex;
```

Comment:
We won't have a global lock on timeline traces with a shared-exclusive lock here. Shared access on tracking and exclusive access in the `Initialize` and `Cleanup`. We can't use `std::shared_mutex` because of iOS `libc++` but we have `fml/synchronization/SharedLock` for that purpose.

Reply:
done